src/main/java/de/fraunhofer/aisec/cpg/passes/VariableUsageResolver.java

```diff
@@ -208,10 +203,8 @@ private void resolveLocalVarUsage(RecordDeclaration currentClass, Node parent, N
         return;
       }
 
-      var refersTo =
-          walker.getDeclarationForScope(
-              parent,
-              v -> !(v instanceof FunctionDeclaration) && v.getName().equals(ref.getName()));
+      Optional<? extends ValueDeclaration> refersTo =
+          Optional.ofNullable(lang.getScopeManager().resolve(ref));
```

Comment:
This is a problem, since this will invoke the regular resolve of the ScopeManager instead of the resolveFunctionPtr a few lines below. This is a problem for functions that have the same name but different parameters.

Reply:
A possible solution would be to integrate the `resolveFunctionPtr` functionality into the `resolve` method of the ScopeManager.